cccl

```diff
@@ -51,6 +51,48 @@ muffle=
 verbose=
 shared_index=-1
 
+# "dumpmachine" is a CC command-line argument to show the compiler's host triplet
+# "cccl" will follow "clang"'s MSVC "dumpmachine" output
+# "dumpmachinefull" is added as well to show the full MSVC version
+# Process the "dumpmachine"/"dumpmachinefull" arguments first, if they exist, and exit gracefully
+for arg in $@; do
+    if [[ $arg =~ ^-dumpmachine(full)?$ ]]; then
+        cl_output_parts=($($prog 2>&1))
+
+        msvc="msvc"
+        if [ "$arg" == "-dumpmachinefull" ]; then
+            msvc="$msvc${cl_output_parts[6]}"
+        fi
+
+        arch="$(echo ${cl_output_parts[8]} | awk '{print tolower($0)}')"
+        output="pc-windows-$msvc"
+        case $arch in
+
+        x64)
+            output="x86_64-$output"
+            ;;
+
+        # "file" command, with "libmagic", shows "Intel 80386" for "X86" architecture
+        x86)
+            output="i386-$output"
+            ;;
+
+        arm64)
+            output="aarch64-$output"
+            ;;
+
+        # "file" command, with "libmagic", shows "ARMv7" for "ARM" architecture
+        arm)
+            output="armv7-$output"
+            ;;
```

Comment:
I'd argue that this should just be "arm" intead of "armv7" using gcc and clang as a reference. Output can be see using [Godbolt ](http://godbolt.org) where the output of `-dumpmachine" for both clang and gcc on armv7 32bit is "arm-unknown-linux-gnueabihf".